components/collector/src/source_collectors/visual_studio_trx/source_up_to_dateness.py

```diff
@@ -0,0 +1,20 @@
+"""Visual Studio TRX source up-to-dateness collector."""
+
+from datetime import datetime
+
+from shared.utils.date_time import now
+
+from base_collectors import TimePassedCollector, XMLFileSourceCollector
+from collector_utilities.date_time import parse_datetime
+from collector_utilities.functions import parse_source_response_xml_with_namespace
+from collector_utilities.type import Response
+
+
+class VisualStudioTRXSourceUpToDateness(XMLFileSourceCollector, TimePassedCollector):
+    """Collector to collect the Visual Studio TRX report age."""
+
+    async def _parse_source_response_date_time(self, response: Response) -> datetime:
+        """Override to parse the timestamp from the response."""
+        tree, namespaces = await parse_source_response_xml_with_namespace(response)
+        times = tree.findall("./ns:Times", namespaces)
```

Comment:
Is it an option to use `tree.find()` to only return the first match?

Reply:
Yeah, that makes more sense.